src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -1512,6 +1528,10 @@ namespace
             ThreadStubArguments* args = static_cast<ThreadStubArguments*>(argument);
             assert(args != nullptr);
 
+#ifdef __APPLE__
+            SetThreadName(GetCurrentThread(), args->name);
```

Comment:
@cshung, I think we can use it on all targets https://github.com/dotnet/runtime/pull/108370?

Reply:
> @cshung, I think we can use it on all targets #108370?

Yes we can, but that would just be extra work since for all other platforms we have already set the name earlier.